libminifi/include/io/ClientSocket.h

```diff
@@ -93,7 +109,7 @@ class Socket : public BaseStream {
    */
   virtual int16_t initialize();
 
-  virtual void setInterface(io::NetworkInterface &&interface) {
+  virtual void setInterface(io::NetworkInterface interface) noexcept {
```

Comment:
I know `noexcept` is rarely used in our codebase but I usually prefer to mark things that do not throw as `noexcept`. The only value is extra documentation and one more register available during code generation, but it's a good default IMO. The downside is if we use it carelessly, exceptions become crashes.

https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#e12-use-noexcept-when-exiting-a-function-because-of-a-throw-is-impossible-or-unacceptable

I'm open to deciding to use another approach, but in that case we need to formalize it. I'd prefer the above.

Reply:
Removed `noexcept`, marking this discussion as resolved. For my arguments on `noexcept`, refer to the thread about `NetworkInterface::operator=(NetworkInterface &&other)`.